### DISCUSSION TITLE
function missing

### DISCUSSION THREAD

**Discussion by ywb2018** [NVIDIA/Megatron-LM#978] [Q&A]
i cannot  find the defination of the function get_blend_from_list , anyone help?

**Comment by Advait251206** [comment]
`get_blend_from_list` does exist in Megatron-LM, but its location depends on the version/commit you're using. The function has been moved during refactoring, so searching in older documentation or different branches may not find it where you expect.

A good way to locate it is to search the repository:

```bash
git grep "def get_blend_from_list"
```

or

```bash
git grep "get_blend_from_list"
```

If the search returns nothing, then you're likely looking at a version where the function has either:

- Been renamed.
- Been moved to another utility module.
- Been removed and its logic inlined into the dataset-building code.

In recent versions of Megatron-LM, dataset blending utilities are typically found under modules such as:

```text
megatron/core/datasets/
```

or

```text
megatron/training/
```

If your code imports it like:

```python
from xxx import get_blend_from_list
```

but the definition is missing, it's also possible that:

- You're using a different branch than the one the documentation refers to.
- Your local repository is incomplete or out of sync.
- The documentation is referring to an older commit where the function still existed.

If you can share:
- the **commit hash** (`git rev-parse HEAD`),
- the **branch** you're using,
- or the file that references `get_blend_from_list`,

it would be much easier to determine where the function went or what replaced it.

---

If this answer helped or pointed you in the right direction, I'd appreciate it if you could mark it as the accepted answer so it's easier for others with the same issue to find.

Also, if you found my contribution useful, I'd appreciate it if you could check out my GitHub profile, follow me, and star any repositories you find interesting.

GitHub: https://github.com/Advait251206
